### PR TITLE
[OPS] Utilize Docker layer caching in PR builds

### PR DESCRIPTION
**Pull request by i5okie**
Take advantage of [`docker/build-push-action@v2`](https://github.com/docker/build-push-action) and [`docker/setup-buildx-action@v1`](https://github.com/docker/setup-buildx-action) GitHub Actions with Caching enabled

### CODE REVIEW

**Review by mtpultz — APPROVED** [review]
🏖 🍹

**Review by james-hollinger — APPROVED** [review]
L Gᶦᵗʰᵘᵇ TM

File: .github/workflows/build-pr.yml [hunk 1]
```diff
@@ -155,24 +186,27 @@ jobs:
...
-          DOCKER_BUILDKIT: 1
-          PR_NUMBER: ${{steps.PR.outputs.number}}
-          REDIRECT_URL: "https://pr-${{steps.PR.outputs.number}}.pharmanetenrolment.gov.bc.ca"
+        uses: docker/build-push-action@v2
+        with:
+          context: prime-angular-frontend
+          push: true
+          file: prime-angular-frontend/openshift.dockerfile
+          tags: ${{secrets.OPENSHIFT_IMAGE_REPOSITORY}}/${{secrets.OPENSHIFT_LICENSE_PLATE}}-tools/prime-frontend:pr-${{steps.PR.outputs.number}}
+          secrets:
```
Comment (itchison) [inline .github/workflows/build-pr.yml:195]: Aren't all of these available as secrets?

File: .github/workflows/build-pr.yml [hunk 2]
```diff
@@ -155,24 +186,27 @@ jobs:
...
+            JWT_WELL_KNOWN_CONFIG="https://dev.oidc.gov.bc.ca/auth/realms/v4mbqqas/.well-known/openid-configuration"
+            KEYCLOAK_URL="https://dev.oidc.gov.bc.ca/auth"
```
Comment (itchison) [inline .github/workflows/build-pr.yml:197]: For example this one.

File: .github/workflows/build-pr.yml (hunk 1 above)
Comment (itchison) [reply .github/workflows/build-pr.yml:195]: See below for a real example.

Comment (i5okie) [reply .github/workflows/build-pr.yml:195]: not quite. these are just build arguments. so should be `build-args` really. I'll correct this.

**Review by AMiKaH — APPROVED** [review]

**Review by itchison — APPROVED** [review]
Thanks for working through my comments! LGTM

**Pull request merged**